airflow/www/views.py

```diff
@@ -1696,10 +1699,13 @@ def graph(self, session=None):
 
         root = request.args.get('root')
         if root:
+            rel = request.args.get('rel')
+            include_upstream = False if rel == "downstream" else True
+            include_downstream = False if rel == "upstream" else True
             dag = dag.sub_dag(
                 task_regex=root,
-                include_upstream=True,
-                include_downstream=False)
+                include_upstream=include_upstream,
+                include_downstream=include_downstream)
```

Comment:
snuck in this code block to allow graph view to use `root` with either upstream, downstream or both. it also changes `root` to use "both" by default.